ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/shader.cpp

```diff
@@ -1046,4 +1045,9 @@ namespace enigma
       it->second.vao = enigma::bound_vbo;
     //}
   }
+
+  void cleanup_shaders() {
+    for(size_t i = 0; i < shaderprograms.size(); ++i) shaderprograms[i] = nullptr;
```

Comment:
Why not just call `shaderprograms.clear()`? Also, the default destructor will do this automatically, so you don't need to call this at all to stop memory leaks.

Reply:
the default destructor calls a gl function. So if I don't clear these before the context is deleted it will cause a crash on exit because all the gl functions are unbound. As for clear() I guess I can do that but gm/enigma typically don't resuse spaces in the vectors so I thought it best to just unassign the array elements to be consistent incase someone else calls this outside the context of an exit

Reply:
Don't forgot that actually happened to us before and I had to spend a nasty time digging it out. I thought I could use a static stack-allocated object destructor to destroy GL objects.

Reply:
We need to revisit our affinity for gaps in the future... for now, I guess this doesn't overly matter.